cli/command/utils_test.go

```diff
@@ -0,0 +1,86 @@
+package command
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestWaitForDeployment(t *testing.T) {
+	defer SetTimeMultiplier(0)()
+
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	deployments := []models.Deployment{
```

Comment:
I think we can use `gomock.InOrder(...)` to make this test simulate a flapping service (see [docs](https://godoc.org/github.com/golang/mock/gomock)):
```
gomock.InOrder(
    mockObj.EXPECT().SomeMethod(1, "first"),
    mockObj.EXPECT().SomeMethod(2, "second"),
    mockObj.EXPECT().SomeMethod(3, "third"),
)
```

Reply:
maybe like:
```
gomock.InOrder(
    // simulate deployments usually taking a while to run
    mockObj.EXPECT().Read(Desired: 1, Running: 0).Times(3),
    // simulate a flapping service
    mockObj.EXPECT().Read(Desired: 1, Running: 1),
    mockObj.EXPECT().Read(Desired: 1, Running: 0).Times(2),
    // simulate the service up and running
    mockObj.EXPECT().Read(Desired: 1, Running: 1).AnyTimes()
)
```

Reply:
I like this a lot. ✅